server/util/live-loan/live-loan-fetch.js

```diff
@@ -197,12 +233,47 @@ const getFilterArrays = filterString => {
 	return [...matches].map(match => [match[1], match[2]]);
 };
 
+// Helper function to find possible filter options from a given list
+const findFilterOption = (options, name, value) => {
+	const option = options.find(o => o?.name?.toLowerCase() === value);
+	if (!option) {
+		warn(`Unknown ${name} "${value}"`);
+	}
+	return option;
+};
+
+// Takes a string like: "sort_expiringSoon,gender_female"
+// and extracts the sort option, e.g. "expiringSoon"
+async function parseSortString(sortString, sortOptions) {
```

Comment:
Does this method need `async`?

Reply:
No, looks like it does not

Reply:
https://github.com/kiva/ui/pull/5479/commits/662bb2adc063e9ca7268ccda98ac6bc94bcc9724